### ISSUE TITLE
Build issues: Intel 2019 and Boost

### ISSUE THREAD

**Issue by marcfehling** [dealii/dealii#7546]
I am currently trying to compile `deal.II` on one of our supercomputers. Using the latest Intel compiler `Intel 19.0.0.117 20180804`, I run into the following `boost`-related issues:

- Compiling with the bundled `boost 1.62.0` version yields: 
  ```
  .../dealii/bundled/boost-1.62.0/include/boost/multi_index_container.hpp(178): internal error: assertion   failed at: "shared/cfe/edgcpfe/scope_stk.c", line 11403

        bfm_allocator(al),
                     ^

  compilation aborted for .../dealii/source/base/parameter_handler.cc (code 4)
  ```
- Compiling with `boost 1.68.0` available on our supercomputer yields:
  ```
  In file included from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/index/detail/rtree/node/node.hpp(14),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/index/rtree.hpp(56),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/algorithms/detail/within/multi_point.hpp(33),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/algorithms/detail/within/implementation.hpp(44),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/algorithms/detail/covered_by/implementation.hpp(26),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/algorithms/covered_by.hpp(24),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/algorithms/detail/distance/point_to_geometry.hpp(42),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/strategies/cartesian/centroid_weighted_length.hpp(25),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/strategies/strategies.hpp(58),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry/geometry.hpp(49),
                   from .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/geometry.hpp(17),
                   from .../dealii/include/deal.II/boost_adaptors/point.h(21),
                   from .../dealii/include/deal.II/boost_adaptors/bounding_box.h(21),
                   from .../dealii/include/deal.II/numerics/rtree.h(23),
                   from .../dealii/include/deal.II/grid/grid_tools.h(39),
                   from .../dealii/source/hp/dof_handler.cc(29):
  .../Boost/1.68.0-iimpi-2018b.1-Python-3.6.6/include/boost/container/vector.hpp(3360): error: argument list for class template "boost::container::vector" is missing
    vector(InputIterator, InputIterator) ->
    ^
  ```
  The latter problem may be related to the `boost` version on our system. I'll investigate further on this issue.

Compiling `deal.II` with `Intel 18.0.2 20180210` and `boost 1.66.0` works fine, though. I suggest to fall back on this combination if you run into the same problems.

**Comment by marcfehling** [comment]
I tried to build `deal.II` with `Intel 19` and the [latest release](https://github.com/boostorg/boost/releases/tag/boost-1.69.0) of `boost 1.69.0`, but that didn't help either.

I'll open an issue in the `boostorg/container` repository.

**Comment by drwells** [comment]
I'm surprised we haven't run into this issue before since its the bundled copy of boost; thanks for reporting here and upstream.

This is only an issue with C++17, right?

**Comment by masterleinad** [comment]
Yes, these problems only appear when compiling with `C++17` support.

**Comment by marcfehling** [comment]
The latter issue should be fixed by the patch I provided for `boost`.

However, compiling with `boost 1.69.0` now also yields to the former one, since it occurs much later in the compilation process. :-(
It seems like this bug [has already been fixed](https://software.intel.com/en-us/forums/intel-c-compiler/topic/733623) in a previous version of the Intel compiler, but seems to have reappeared. I'll try to compile the minimal example and write a report to Intel if the error remains.
